client/src/components/Filters/PriceFilter.js

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+import ToggleButton from "@mui/material/ToggleButton";
+import ToggleButtonGroup from "@mui/material/ToggleButtonGroup";
+
+const DEFAULT_FILTER = {
+  $: false,
+  $$: false,
+  $$$: false,
+  $$$$: false,
+};
+
+function PriceCategoryFilter(restaurant) {
+  return selectedPrices.includes(restaurant.price_category);
+}
+
+export default function PriceFilterToggler() {
+  const [priceFilter, setPriceFilter] = React.useState(DEFAULT_FILTER);
+  const [filters, setFilters] = React.useState(() => []);
+
+  const handleFormat = (event, newFormats) => {
+    setFilters(newFormats);
+  };
+  const selectedPrices = Object.keys(priceFilter).filter(
+    (priceKey) => priceFilter[priceKey]
+  );
+  return (
+    <ToggleButtonGroup
+      value={filters}
+      onChange={handleFormat}
+      aria-label="price"
+      size="small"
+    >
+      <ToggleButton
+        value="$"
+        onClick={() => {
+          const updatedFilter = {
+            ...priceFilter,
+            $: !priceFilter["$"],
+          };
+          setPriceFilter(updatedFilter);
+          console.log(updatedFilter);
+        }}
+      >
+        $
+      </ToggleButton>
+      <ToggleButton
+        value="$$"
+        onClick={() => {
+          const updatedFilter = {
+            ...priceFilter,
+            $$: !priceFilter["$$"],
+          };
+          setPriceFilter(updatedFilter);
+          console.log(updatedFilter);
+        }}
+      >
+        $$
+      </ToggleButton>
+      <ToggleButton
+        value="$$$"
+        onClick={() => {
+          const updatedFilter = {
+            ...priceFilter,
+            $$$: !priceFilter["$$$"],
+          };
+          setPriceFilter(updatedFilter);
+          console.log(updatedFilter);
+        }}
+      >
+        $$$
+      </ToggleButton>
+      <ToggleButton
+        value="$$$$"
+        onClick={() => {
+          const updatedFilter = {
+            ...priceFilter,
+            $$$$: !priceFilter["$$$$"],
+          };
+          setPriceFilter(updatedFilter);
+          console.log(updatedFilter);
+        }}
+      >
+        $$$$
+      </ToggleButton>
```

Comment:
This works, but I dont like there's a lot of repeated code here. Can you think of a way to automate this?
Hint: you can have an array containing : `['$', '$$', '$$$', '$$$$']` and create a button for each category using `array.map`.

Also, remove the console logs.